File: Kadane's/Maximum Absolute Sum of Any Subarray.go
func maxAbsoluteSum(nums []int) int {
    curr_max := 0
    curr_min := 0
    kadane_max := 0
    kadane_min := 0

    for _, val := range nums {
        curr_max = max(0, curr_max+val)
        curr_min = min(0, curr_min+val)

        kadane_max = max(kadane_max, curr_max)
        kadane_min = min(kadane_min, curr_min)
    }

    return max(kadane_max, -kadane_min)
}

File: Kadane's/Maximum Absolute Sum of Any Subarray.rs
impl Solution {
    pub fn max_absolute_sum(nums: Vec<i32>) -> i32 {
        let mut kadane_max = 0;
        let mut kadane_min = 0;
        let mut curr_max = 0;
        let mut curr_min = 0;

        for i in 0..nums.len() {
            curr_max = 0.max(curr_max + nums[i]);
            kadane_max = kadane_max.max(curr_max);

            curr_min = 0.min(curr_min + nums[i]);
            kadane_min = kadane_min.min(curr_min);
        }
        kadane_min.abs().max(kadane_max)
    }
}